strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/repository/RepositoryLayoutEnum.java

```diff
@@ -14,7 +14,9 @@
 
     RPM("RPM"),         // Unsupported
 
-    YUM("YUM");         // Unsupported
+    YUM("YUM"),         // Unsupported
+    
+    NUGET_HIERACHLICAL("Nuget Hierarchical");         
```

Comment:
I think this class should actually just get deprecated. It's used only in the [Repository](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/repository/Repository.java).

[This line](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/repository/Repository.java#L47):

    private String layout = RepositoryLayoutEnum.MAVEN_2.getLayout();

Should now be changed to just:

    private String layout = Maven2LayoutProvider.ALIAS;

And the whole `RepositoryLayoutEnum` could go, making things much more dynamic... I mean... When someone else decides to add a new `LayoutProvider`, they won't have to change this class and so on.